services/worker-manager/src/providers/gcpprovider.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+const {Bid} = require('../bid');
+const {Provider} = require('../provider');
+const Worker = require('../worker');
+const Compute = require('@google-cloud/compute');
+
+class GCPProvider extends Provider {
+
+  /**
+   * Construct a GCP Provider
+   */
+  constructor({id}) {
+    super({id});
+    this.gcpCompute = new Compute();
+  }
+
+  convertToGCPStatus(taskclusterWorkerState) {
+    switch (taskclusterWorkerState) {
+      case Provider.states.booting: return ['STAGING'];
+      case Provider.states.requested: return ['PROVISIONING'];
+      case Provider.states.running: return ['RUNNING'];
+      case Provider.states.terminating: return ['STOPPING', 'SUSPENDING', 'SUSPENDED', 'TERMINATED'];
+    }
+  }
+
+  convertFromGCPStatus(gcpInstanceStatus) {
+    switch (gcpInstanceStatus) {
+      case 'PROVISIONING': return Provider.states.requested;
+      case 'STAGING': return Provider.states.requested;
+      case 'RUNNING': return Provider.states.running;
+      case 'STOPPING': return Provider.states.terminating;
+      case 'SUSPENDING': return Provider.states.terminating;
+      case 'SUSPENDED': return Provider.states.terminating;
+      case 'TERMINATED': return Provider.states.terminating;
+    }
+  }
+
+  /**
+   * this is called in provisioner
+   *
+   * @param states Array<String>
+   * @param workerTypes Array<String>
+   * @returns {Promise<Array<Worker>>}
+   */
+  async listWorkers({states, workerTypes}) {
+    console.log(`Got states: ${states}, and worker types: ${workerTypes}`);
+
+    let filter = workerTypes.map(wt => `(labels.worker-type = ${wt})`).join(' OR ');
+
+    const instances = (await this.gcpCompute.getVMs({
+      autoPaginate: false,
```

Comment:
What does this do?

Reply:
oh, I actually have to remove this parameter. It turns off automatic pagination handling. When I started with this, I was hoping that would turn pagination off, but it seems it just switches that to manual mode.

Speaking of pagination, I have to add some code here to account for pagination